subsys/disk/disk_access_sdhc.h

```diff
@@ -0,0 +1,500 @@
+/* disc_access_sdhc.h - header SDHC*/
+
+/*
+ * Copyright (c) 2019 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DISK_ACCESS_SDHC_H_
+#define ZEPHYR_INCLUDE_DISK_ACCESS_SDHC_H_
+
+#include <spi.h>
+
+#define SDMMC_CLOCK_400KHZ (400000U)
+#define SD_CLOCK_25MHZ (25000000U)
+#define SD_CLOCK_50MHZ (50000000U)
+#define SD_CLOCK_100MHZ (100000000U)
+#define SD_CLOCK_208MHZ (208000000U)
+#define MMC_CLOCK_26MHZ (26000000U)
+#define MMC_CLOCK_52MHZ (52000000U)
+#define MMC_CLOCK_DDR52 (52000000U)
+#define MMC_CLOCK_HS200 (200000000U)
+#define MMC_CLOCK_HS400 (400000000U)
+
+/* Command IDs */
+#define SDHC_GO_IDLE_STATE 0
+#define SDHC_ALL_SEND_CID 2
+#define SDHC_SEND_RELATIVE_ADDR 3
+#define SDHC_SWITCH 6
+#define SDHC_SELECT_CARD 7
+#define SDHC_SEND_IF_COND 8
+#define SDHC_SEND_CSD 9
+#define SDHC_SEND_CID 10
+#define SDHC_VOL_SWITCH 11
+#define SDHC_STOP_TRANSMISSION 12
+#define SDHC_SEND_STATUS 13
+#define SDHC_GO_INACTIVE_STATE 15
+#define SDHC_SET_BLOCK_SIZE 16
+#define SDHC_READ_SINGLE_BLOCK 17
+#define SDHC_READ_MULTIPLE_BLOCK 18
+#define SDHC_SEND_TUNING_BLOCK 19
+#define SDHC_SET_BLOCK_COUNT 23
+#define SDHC_WRITE_BLOCK 24
+#define SDHC_WRITE_MULTIPLE_BLOCK 25
+#define SDHC_ERASE_BLOCK_START 32
+#define SDHC_ERASE_BLOCK_END 33
+#define SDHC_ERASE_BLOCK_OPERATION 38
+#define SDHC_APP_CMD 55
+#define SDHC_READ_OCR 58
+#define SDHC_CRC_ON_OFF 59
+
+enum sdhc_app_ext_cmd {
+	SDHC_APP_SET_BUS_WIDTH = 6,
+	SDHC_APP_SEND_STATUS = 13,
+	SDHC_APP_SEND_NUM_WRITTEN_BLK = 22,
+	SDHC_APP_SET_WRITE_BLK_ERASE_CNT = 23,
+	SDHC_APP_SEND_OP_COND = 41,
+	SDHC_APP_CLEAR_CARD_DETECT = 42,
+	SDHC_APP_SEND_SCR = 51
+};
+
+#define SDHC_SEND_OP_COND SDHC_APP_SEND_OP_COND
+
+/* R1 response status */
+#define SDHC_R1_IDLE 0x01
+#define SDHC_R1_ERASE_RESET 0x02
+#define SDHC_R1_ILLEGAL_COMMAND 0x04
+#define SDHC_R1_COM_CRC 0x08
+#define SDHC_R1_ERASE_SEQ 0x10
+#define SDHC_R1_ADDRESS 0x20
+#define SDHC_R1_PARAMETER 0x40
+
+#define SDHC_SECTOR_SIZE 512
```

Comment:
Not sure we should have this parameter as a macro, rather set it in Kconfig.
This is the only parameter I've noticed, there may be others.

Reply:
Hi,
http://users.ece.utexas.edu/~valvano/EE345M/SD_Physical_Layer_Spec.pdf 7.3.2.1 Format R1

Reply:
How is section 7.3.2.1 a response to SDHC_SECTOR_SIZE?

Reply:
define SDHC_SECTOR_SIZE as 512B, which is the smallest sector size of SD. To get better R/W performance, we can set sector size by CMD16 to multiple 512B.